quodlibet/quodlibet/ext/songsmenu/tapbpm.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Didier Villevalois,
+#           2010 Christoph Reiter
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation
+
+from gi.repository import Gtk, GObject
+
+from quodlibet import _
+from quodlibet.plugins.songsmenu import SongsMenuPlugin
+from quodlibet.qltk import Icons
+from quodlibet.qltk.window import Dialog
+
+
+class TapBpmPanel(Gtk.VBox):
+    def __init__(self, parent, song):
+        super(Gtk.VBox, self).__init__()
+
+        self.dialog = parent
+        self.song = song
+        self.original_bpm = float(song["bpm"]) if "bpm" in song else 0.0
+
+        self.set_margin_bottom(6)
+        self.set_spacing(6)
+
+        box = Gtk.HBox()
+        box.set_spacing(6)
+        box.pack_start(Gtk.Label(_("BPM:")), False, True, 0)
+        self.bpm_label = Gtk.Label(_("n/a"))
+        self.bpm_label.set_xalign(0.5)
+        box.pack_start(self.bpm_label, True, True, 0)
+
+        self.reset_btn = Gtk.Button(label=_("Reset"))
+        self.reset_btn.connect('clicked', lambda *x: self.reset())
+        box.pack_end(self.reset_btn, False, True, 0)
+
+        self.pack_start(box, False, True, 0)
+
+        self.tap_btn = Gtk.Button(label=_("Tap"))
+        self.tap_btn.connect('button-press-event', self.tap)
```

Comment:
'clicked' event, otherwise pressing space/enter wont do anything if the button is focused.